test/func/sproxyd_utils.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) 2014 Scality
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import with_statement
+import os
+import platform
+import subprocess32
+import urllib
+
+
+ring_driver_conf_tmpl = """"ring_driver:0": {
+        "by_path_service_id": "%(by_path_service_id)s",
+        "by_path_cos": 0,
+        "alias": "chord_path",
+        "bstraplist": "%(bootstraplist)s",
+        "by_path_enabled": true,
+        "deferred_deletes_enabled_by_policy": false,
+        "deferred_deletes_enabled_by_request": false,
+        "deferred_writes_enabled_by_policy": false,
+        "deferred_writes_enabled_by_request": false,
+        "type": "chord"
+     }"""
+
+sproxyd_conf_tmpl = """{
+    "general": {
+        "ring": "MyRing",
+        "conn_max": 10000,
+        "conn_max_reuse": 100000,
+        "consistent_reads": true,
+        "consistent_writes": true,
+        "max_proc_fd": 40960,
+        "port": %(fcgi_port)s,
+        "split_chunk_size": 33554432,
+        "split_control_by_request": false,
+        "split_enabled": true,
+        "split_gc_cos": 2,
+        "split_memory_limit": 671088640,
+        "split_n_get_workers": 20,
+        "split_n_io_workers": 20,
+        "split_n_put_workers": 20,
+        "split_threshold": 67108864
+    },
+    %(ring_driver)s
+}"""
+
+
+apache_vhost_conf_tmpl = """Listen %(vhost_port)s
+    <VirtualHost *:%(vhost_port)s>
+    DocumentRoot /var/www/%(name)s
+    AllowEncodedSlashes %(allow_encoded_slash)s
+    LimitRequestFieldSize 32766
+    LimitRequestLine 32766
+    </VirtualHost>
+    FastCgiExternalServer /var/www/%(name)s/proxy -host 127.0.0.1:%(fcgi_port)s -flush -idle-timeout 300
+    """
+
+
+class SproxydProcess(object):
+
+    def __init__(self, name, conf_file):
+        self.name = name
+        self.conf_file = conf_file
+        self._p = None
+
+    def __getattr__(self, attr):
+        return getattr(self._p, attr)
+
+    def start(self):
+        self._f = open('/opt/stack/logs/%s.out' % self.name, 'a')
+        self._p = subprocess32.Popen(
+            ['sudo', '/usr/local/bin/sproxyd',  '-wdl', '-c', '%s' %
+             self.conf_file, '-n', self.name,
+             '-TLERROR,LWARNING,LRINGLL,LRINGLL'],
```

Comment:
Could you leave a comment on what these tokens do ? For instance, I don't know what LRINGLL means, but now I see double LRINGLL so I feel it's important but I don't know what this is about.